code/game/objects/items/melee/baton.dm

```diff
@@ -0,0 +1,656 @@
+/obj/item/melee/baton
+	name = "police baton"
+	desc = "A wooden truncheon for beating criminal scum. Left click to stun, right click to harm."
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "classic_baton"
+	inhand_icon_state = "classic_baton"
+	worn_icon_state = "classic_baton"
+	lefthand_file = 'icons/mob/inhands/equipment/security_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/security_righthand.dmi'
+	slot_flags = ITEM_SLOT_BELT
+	force = 12 //9 hit crit
+	w_class = WEIGHT_CLASS_NORMAL
+	wound_bonus = 15
+
+	/// Whether this baton is active or not
+	var/active = TRUE
+	/// Used interally, you don't want to modify
+	var/cooldown_check = 0
+	/// Default wait time until can stun again.
+	var/cooldown = (4 SECONDS)
+	/// The length of the knockdown applied to a struck living, non-cyborg mob.
+	var/knockdown_time = (1.5 SECONDS)
+	/// If affect_cyborg is TRUE, this is how long we stun cyborgs for on a hit.
+	var/stun_time_cyborg = (5 SECONDS)
+	/// The length of the knockdown applied to the user on clumsy_check()
+	var/clumsy_knockdown_time = 18 SECONDS
+	/// How much stamina damage we deal on a successful hit against a living, non-cyborg mob.
+	var/stamina_damage = 55
+	/// Can we stun cyborgs?
+	var/affect_cyborg = FALSE
+	/// The path of the default sound to play when we stun something.
+	var/on_stun_sound = 'sound/effects/woodhit.ogg'
+	/// The volume of the above.
+	var/on_stun_volume = 75
+	/// Do we animate the "hit" when stunning something?
+	var/stun_animation = TRUE
+	/// Whether the stun attack is logged. Only relevant for abductor batons, which have different modes.
+	var/log_stun_attack = TRUE
+
+/obj/item/melee/baton/Initialize()
+	. = ..()
+	// Adding an extra break for the sake of presentation
+	if(stamina_damage != 0)
+		offensive_notes = "\nVarious interviewed security forces report being able to beat criminals into exhaustion with only [span_warning("[round(100 / stamina_damage, 0.1)] hit\s!")]"
```

Comment:
You should instead use ceiling to round up because that'll be the amount of hits you need to stun a criminal.

Reply:
make it use the random victim name instead of criminals